saltproc/openmc_depcode.py

```diff
@@ -0,0 +1,326 @@
+    template files for running OpenMC depletion simulations.
+    Also contains neutrons population, active, and inactive cycles.
+    Contains methods to read template and output files,
+    write new input files for OpenMC.
+
+    Attributes
+    ----------
+    param : dict of str to type
+        Holds depletion step parameter information. Parameter names are keys
+        and parameter values are values.
+    sim_info : dict of str to type
+        Holds simulation settings information. Setting names are keys
+        and setting values are values.
+    iter_inputfile : dict of str to str
+        Paths to OpenMC input files for OpenMC rerunning.
+    iter_matfile : str
+        Path to iterative, rewritable material file for OpenMC
+        rerunning. This file is modified during the simulation.
+
+
+
+    """
+
+    def __init__(self,
+                 exec_path="openmc_deplete.py",
+                 template_input_file_path={"geometry": "./geometry.xml",
+                                           "settings": "./settings.xml",
+                                           "chain_file": "./chain_simple.xml"},
+                 geo_files=None,
+                 npop=50,
+                 active_cycles=20,
+                 inactive_cycles=20):
+        """Initializes the OpenMCDepcode object.
+
+           Parameters
+           ----------
+           exec_path : str
+               Path to OpenMC depletion script.
+           template_input_file_path : dict of str to str
+               Path to user input files (``.xml`` file for geometry,
+               material, and settings) for OpenMC. File type as strings
+               are keys (e.g. 'geometry', 'settings', 'material'), and
+               file path as strings are values.
+           geo_files : str or list, optional
+               Path to file that contains the reactor geometry.
+               List of `str` if reactivity control by
+               switching geometry is `On` or just `str` otherwise.
+           npop : int, optional
+               Size of neutron population per cycle for Monte Carlo.
+           active_cycles : int, optional
+               Number of active cycles.
+           inactive_cycles : int, optional
+               Number of inactive cycles.
+
+        """
+        super().__init__("openmc",
+                         exec_path,
+                         template_input_file_path,
+                         geo_files,
+                         npop,
+                         active_cycles,
+                         inactive_cycles)
+        self.iter_inputfile = {'geometry': './geometry.xml',
+                               'settings': './settings.xml'},
+        self.iter_matfile = './materials.xml'
+
+    def read_depcode_info(self):
+        """Parses initial OpenMC simulation info from the OpenMC output files
+        and stores it in the `Depcode` object's ``sim_info`` attribute.
+        """
```

Comment:
I'm assuming that most of the functions are placeholders for now, correct?

Reply:
This actually confuses me, too. These are no longer `@abstractmethods` so shouldn't they do something?

Reply:
These are placeholders and will be implemented in a future PR in the near future. Right now these methods don't actually do anything.

Reply:
These are placeholders and will be implemented in a future PR in the near future. Right now these methods don't actually do anything.